ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -353,15 +353,15 @@ public void onResponse(Call call, Response response) throws IOException {
             try {
               // If JS wants progress updates during the download, and it requested a text response,
               // periodically send response data updates to JS.
-              if (useIncrementalUpdates && responseType.equals("text")) {
+              if (useIncrementalUpdates && responseType != null && responseType.equals("text")) {
                 readWithProgress(eventEmitter, requestId, responseBody);
                 ResponseUtil.onRequestSuccess(eventEmitter, requestId);
                 return;
               }
 
               // Otherwise send the data in one big chunk, in the format that JS requested.
               String responseString = "";
-              if (responseType.equals("text")) {
+              if (responseType != null && responseType.equals("text") && responseBody != null) {
                 responseString = responseBody.string();
               } else if (responseType.equals("base64")) {
```

Comment:
We have to check `responseType` nullability here as well. Or, even better, we can check nulabity somewhere above.